view/frontend/web/js/view/checkout/summary/agreements-assigner-minix.js

```diff
@@ -0,0 +1,51 @@
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2021 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+/*global alert*/
+define([
+    'mage/utils/wrapper',
+    'jquery'
+
+], function (wrapper, $) {
+    'use strict';
+
+    return function (agreementsAssignerAction) {
+        return wrapper.wrap(agreementsAssignerAction, function (originalAction, paymentData) {
+            originalAction(paymentData);
+            var agreementIds = paymentData['extension_attributes']['agreement_ids'];
+            if (paymentData.additional_data && paymentData.additional_data.stateData) {
+                let data = paymentData['additional_data']['stateData'];
+                let stateData = JSON.parse(data);
+                if (stateData.paymentMethod.type == 'amazonpay' && !agreementIds.length) {
+                    agreementIds = ["1"];
```

Comment:
This would break when the agreement ID is different than 1, or when there's more than 1 agreement enabled. Ideally these IDs would come from somewhere else. How is it retrieved in the original `Magento_CheckoutAgreements/js/model/agreements-assigner`?

Reply:
Indeed this is something to check

Reply:
It is switched to `window.checkoutConfig`. This includes the agreement details so we are able to get the length of the array